File: ts-plugins/src/testing/scenario-generation-plugin.ts
import type { FlowDefinition } from '@unlaxer/tramli';
import type { GenerationPlugin, PluginDescriptor } from '../api/types.js';
import type { FlowTestPlan } from './types.js';
import { ScenarioTestPlugin } from './scenario-test-plugin.js';

export class ScenarioGenerationPlugin<S extends string> implements GenerationPlugin<FlowDefinition<S>, FlowTestPlan> {
  private readonly delegate = new ScenarioTestPlugin();

  descriptor(): PluginDescriptor {
    return {
      id: 'scenario-tests',
      displayName: 'Scenario Test Generator',
      description: 'Produces scenario-oriented test plans from a flow definition.',
    };
  }
  kind() { return 'GENERATION' as const; }

  generate(input: FlowDefinition<S>): FlowTestPlan {
    return this.delegate.generate(input);
  }
}

File: lang/java-plugins/src/main/java/org/unlaxer/tramli/plugins/testing/ScenarioGenerationPlugin.java
package org.unlaxer.tramli.plugins.testing;

import org.unlaxer.tramli.FlowDefinition;
import org.unlaxer.tramli.FlowState;
import org.unlaxer.tramli.plugins.api.GenerationPlugin;
import org.unlaxer.tramli.plugins.api.PluginDescriptor;

public final class ScenarioGenerationPlugin<S extends Enum<S> & FlowState> implements GenerationPlugin<FlowDefinition<S>, FlowTestPlan> {
    private final ScenarioTestPlugin delegate = new ScenarioTestPlugin();

    @Override
    public PluginDescriptor descriptor() {
        return new PluginDescriptor("scenario-tests", "Scenario Test Generator", "Produces scenario-oriented test plans from a flow definition.");
    }

    @Override
    public FlowTestPlan generate(FlowDefinition<S> input) {
        return delegate.generate(input);
    }
}
